src/app/layout/header/dialog-user/dialog-user.component.ts

```diff
@@ -1,102 +1,72 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { FormBuilder, FormGroup, Validators } from '@angular/forms';
 import { MatDialogRef } from '@angular/material/dialog';
-import { ToastrService } from 'ngx-toastr';
-import { IamService } from 'src/app/shared/services/iam.service';
-import { LoadingService } from 'src/app/shared/services/loading.service';
-import { CancelButton } from '../../loading/loading.component';
+import { Store } from '@ngrx/store';
+import * as userSelectors from '../../../state/user-claim/user.selectors';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import * as userActions from '../../../state/user-claim/user.actions';
+import { UserClaimState } from '../../../state/user-claim/user.reducer';
 
 @Component({
-    selector: 'dialog-user',
-    templateUrl: 'dialog-user.component.html',
-    styleUrls: ['../header.component.scss']
+  selector: 'app-dialog-user',
+  templateUrl: 'dialog-user.component.html',
+  styleUrls: ['../header.component.scss']
 })
-export class DialogUser implements OnInit {
+export class DialogUserComponent implements OnInit, OnDestroy {
 
-    public profileForm      : FormGroup;
-    public maxDate          : Date;
-    private profileCache: any;
+  public profileForm: FormGroup;
+  public maxDate: Date;
+  private destroy$ = new Subject<void>();
 
-    constructor(
-        public dialogRef: MatDialogRef<DialogUser>,
-        private fb: FormBuilder,
-        private iamService: IamService,
-        private toastr: ToastrService,
-        private loadingService: LoadingService) {
-            this.profileForm = fb.group({
-                name: ['', Validators.compose([
-                    Validators.maxLength(256),
-                    Validators.required
-                ])],
-                birthdate: ['', Validators.required],
-                address: ['', Validators.compose([
-                    Validators.maxLength(500),
-                    Validators.required
-                ])]
-            });
-            
-            let today = new Date();
-            today.setFullYear(today.getFullYear() - 18);
-            this.maxDate = today;    
-    }
-
-    async ngOnInit() {
-        this.loadingService.show();
+  constructor(
+    public dialogRef: MatDialogRef<DialogUserComponent>,
+    private fb: FormBuilder,
+    private store: Store<UserClaimState>) {
+    this.profileForm = fb.group({
```

Comment:
we need to avoid complex actions in the constructor

I think, it will be better:
export class DialogUserComponent implements OnInit, OnDestroy {   
.....
 public profileForm = this.fb.group({
......

Reply:
Yeah, i didn't wrote that. And i don't want to refactor everything at once. However this is small change so i will do that now.